src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx

```diff
@@ -133,7 +133,9 @@ export function DocTableLegacy(renderProps: DocTableLegacyProps) {
       });
     } else if (scope && scope.current) {
       scope.current.renderProps = { ...renderProps, rows, minimumVisibleRows };
-      scope.current.$apply();
+      if (scope.current.$root.$$phase !== '$apply' && scope.current.$root.$$phase !== '$digest') {
```

Comment:
dear @flash1293, god of inner angular, this solves an issue when adding a filter from the classic table, but I'm not sure what change causes this error? could you have a look if this solution is ok? 

Error message:

```
Error: [$rootScope:inprog] $apply already in progress
https://errors.angularjs.org/1.8.0/$rootScope/inprog?p0=%24apply
    at angular.js:138
    at beginPhase (angular.js:19826)
    at ChildScope.$apply (angular.js:19558)
    at create_doc_table_react.tsx:112
    at commitHookEffectList (react-dom.development.js:22030)
    at commitPassiveHookEffects (react-dom.development.js:22064)
    at HTMLUnknownElement.callCallback (react-dom.development.js:336)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:385)
    at invokeGuardedCallback (react-dom.development.js:440)
    at flushPassiveEffectsImpl (react-dom.development.js:25392)
```

video:


https://user-images.githubusercontent.com/463851/119616429-e3705780-be00-11eb-9ac8-fef835033960.mp4

Reply:
not sure what exactly is going on here either (didn't manage to trace it), but I agree with `evalAsync` being the safer alternative.

Reply:
done!

Reply:
Ok, didn't work with $evalAsync, but it worked with `$applyAsync`